docs/learn/expert-guides/key-manager/grant-permissions.md

```diff
@@ -8,464 +8,252 @@ import TabItem from '@theme/TabItem';
 
 # Grant Permissions
 
-:::caution
+This guide will teach you how to grant [Key Manager Permissions](../../../standards/universal-profile/lsp6-key-manager.md#address-permissions) to third-party addresses. Permissioned addresses become _Universal Profile Controllers_ with **individual permissions** to **interact with or on behalf of the Universal Profile**.
 
-This article is a WIP
+:::info Code Examples
+
+The full code of this example can be found in the 👾 [lukso-playground](https://github.com/lukso-network/lukso-playground/tree/main/key-manager) repository.
 
 :::
 
-:::info Requirements
+:::note Requirements
 
-You will need a Universal Profile that you can control via its KeyManager to follow this guide. <br/>
-If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](../universal-profile/create-profile.md) or look at the [_lsp-factory.js_](../../../tools/lsp-factoryjs/deployment/universal-profile.md) docs.
+If you don't have a Universal Profile yet, please follow our previous guide to [**Create a Universal Profile**](../universal-profile/create-profile.md). To follow this guide, you must have a Universal Profile and related controller with the [`ADDCONTROLLER`](../../../standards/universal-profile/lsp6-key-manager#permissions) permission.
```

Comment:
The second sentence might fit better for the context further below, at the point in the code where we set the permissions via `setData` maybe?

Otherwise it is a bit overloaded of information straight away